### PR TITLE
Skip mem index for items that cannot be indexed

### PR DESCRIPTION
**Pull request by nb-ohad**
### Explain the changes
- Where the value for the index key is missing we place a warning and skip

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/server/system_services/system_store.js [hunk 1]
```diff
@@ -331,7 +331,12 @@ class SystemStoreData {
         _.each(COLLECTIONS, col => {
             _.each(col.mem_indexes, index => {
                 _.each(this[col.name], item => {
-                    let key = _.get(item, index.key || '_id').valueOf();
+                    const field = _.get(item, index.key || '_id');
+                    if (!field) {
+                        dbg.warn(`SystemStoreData: Item ${col.name}[${item._id}] could not be indexed for ${index.name}
```
Comment (nimrod-becker) [inline src/server/system_services/system_store.js]:
```suggestion
                        dbg.error(`SystemStoreData: Item ${col.name}[${item._id}] could not be indexed for ${index.name}
```

File: src/server/system_services/system_store.js [hunk 2]
```diff
@@ -331,7 +331,12 @@ class SystemStoreData {
...
+                            (Index field ${index.key} not found on item), skipping`);
+                    }
+                    const key = field.valueOf();
```
Comment (nimrod-becker) [inline src/server/system_services/system_store.js]: means we'd add map[undefined] ? do we want it ?

File: src/server/system_services/system_store.js (hunk 1 above)
Comment (nb-ohad) [reply src/server/system_services/system_store.js]: Changed it to error

File: src/server/system_services/system_store.js (hunk 2 above)
Comment (nb-ohad) [reply src/server/system_services/system_store.js]: Added missing return

**Review by nimrod-becker — APPROVED** [review]

**Pull request merged**